syntax = "proto3";

package perper.grpc2;

import "google/protobuf/descriptor.proto";
import "google/protobuf/empty.proto";
import "google/protobuf/type.proto";

option csharp_namespace = "Perper.Protocol.Protobuf2";
option go_package = "github.com/obecto/perper/scaler/proto2";
option java_multiple_files = true;
option java_package = "com.obecto.perper.protobuf2";

// TODO: reserve a range in https://github.com/protocolbuffers/protobuf/blob/main/docs/options.md
extend google.protobuf.FieldOptions {
  bool first_message_only = 55540;
}
extend google.protobuf.EnumValueOptions {
  string execution_delegate = 54655;
}

service FabricProtobufDescriptors {
  rpc Register(FabricProtobufDescriptorsRegisterRequest) returns (google.protobuf.Empty) {}

  rpc Get(FabricProtobufDescriptorsGetRequest) returns (FabricProtobufDescriptorsGetResponse) {}
}

message FabricProtobufDescriptorsRegisterRequest {
  string type_url = 1;
  oneof descriptor {
    google.protobuf.Type type = 2;
    google.protobuf.Enum enum = 3;
  }
}

message FabricProtobufDescriptorsGetRequest {
  string type_url = 1;
}

message FabricProtobufDescriptorsGetResponse {
  oneof descriptor {
    google.protobuf.Type type = 2;
    google.protobuf.Enum enum = 3;
  }
}
